openapi/openapi_test.go

```diff
@@ -0,0 +1,215 @@
+package openapi
+
+import (
+	"path"
+	"reflect"
+	"strings"
+	"testing"
+)
+
+var openapiFile = "fixtures/validation.yaml"
+var expectedPathInFile = "/endpoint"
+var expectedBasePath = path.Dir(openapiFile)
+
+func TestOpenAPI_LoadOpenAPI(t *testing.T) {
+	openapi, err := LoadOpenAPI(openapiFile)
+	if err != nil {
+		t.Errorf("could not load file %s:%s", openapiFile, err.Error())
+		t.Fail()
+	}
+	if openapi == nil {
+		t.Errorf("could not load file %s:%s", openapiFile, err.Error())
+		t.Fail()
+	}
+}
+
+func TestOpenAPI_SetRenderer(t *testing.T) {
+	openapi, _ := LoadOpenAPI(openapiFile)
+	SetRenderer(openapi, GinRenderer{})
+
+	//Check *PathItem
+	paths := openapi.getChildren()
+	for key, path := range paths {
+		p := path.(*PathItem)
+		if p.getRenderer() == nil {
+			t.Errorf("Path %v is missing a renderer", key)
+			t.Fail()
+		}
+	}
+
+	//Check *Operation
+	operations := paths["/endpoint"].(*PathItem).getChildren()
+	// TODO currently faiing
+	for key, operation := range operations {
+		if operation != nil {
+			o := operation.(*Operation)
+			if o.getRenderer() == nil {
+				t.Errorf("Operation %v is missing a renderer", key)
+				t.Fail()
+			}
+		}
+	}
+
+	//Check *RequestBody
+	requestBodyAndResponses := operations["post"].(*Operation).getChildren()
```

Comment:
getChildren should return parameters as well. This name might become misleading.